connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -375,6 +375,10 @@ private boolean sendRecords() {
                                 // executeFailed here allows the use of existing logging infrastructure/configuration
                                 retryWithToleranceOperator.executeFailed(Stage.KAFKA_PRODUCE, WorkerSourceTask.class,
                                         preTransformRecord, e);
+
+                                //Make sure to clear out the record and update metrics
+                                submittedRecords.removeLastOccurrence(submittedRecord);
```

Comment:
Actually, thinking about this a little more, should this record be removed, or acknowledged?

The reason we added the `SubmittedRecords::removeLastOccurrence` method is that we want to be able to re-submit the same record while handling retriable failures in `WorkerSourceTask::sendRecords`.

If we're not planning on re-submitting this record (which I don't believe we are), and we're trying to skip over the record permanently (which I believe we are), why not just allow its offsets to be committed?

Reply:
Yeah I think you are right we can just ack instead of removing. It's a failed record so it wouldn't be retried, if should be skipped. I can change that and also take a look at making test changes as well that you commented on.